DASH-IF-IOP.bs.md

```diff
@@ -10,9 +10,1068 @@ Live discussion in #document-authoring on Slack.
+
+* Lack of an in-band MPD validity event in a [=media segment=] indicates that an MPD that was valid at the start of the [=media segment=] remains valid up to the end of the [=media segment=] (mapped to wall clock time).
+* The presence of an in-band MPD validity event in a [=media segment=] indicates that the MPD with `MPD@publishTime` equal to the event's `publish_time` field remains valid up to the event start time (mapped to wall clock time).
+
+The in-band events used for signaling MPD validity duration SHALL have `scheme_id_uri` and `value` matching the `InbandEventStream` element. Clients SHALL NOT use in-band events for MPD validity update signaling if these fields on the events do not match the `InbandEventStream` element or if the `InbandEventStream` element is not present.
+
+In-band events with `value=3` SHALL provide an updated MPD in the event's `mpd` field as UTF-8 encoded text without a byte order mark.
+
+Multiple [=media segments=] MAY signal the same validity update event (identified by matching `id` field on event), enabling the signal to be delivered several segments in advance of the MPD expiration.
+
+#### End of live content #### {#timing-mpd-updates-theend}
+
+Live services can reach a point where no more content will be produced - existing content will be played back by clients and once they reach the end, playback will cease.
+
+Services SHALL define a fixed duration for the last [=period=], remove the `MPD@minimumUpdatePeriod` attribute and cease performing MPD updates to signal that no more content will be added to the MPD. The `MPD@type` MAY be changed to `static` at this point or later if the service is to be converted to a static MPD for on-demand viewing.
+
+## XLink elements ## {#timing-xlink}
+
+Some XML elements in an MPD may be external to the MPD itself, delay-loaded by clients based on different triggers. This mechanism is called XLink and it enables client-side MPD composition from different sources. For the purposes of timing and addressing, it is important to ensure that the duration of each [=period=] can be accurately determined both before and after XLink resolution.
+
+Note: XLink functionality in DASH is defined by [[!MPEGDASH]] and [[!XLINK]]. This document provides a high level summary of the behavior and defines interoperability requirements.
+
+XLink elements are those in the MPD that carry the `xlink:href` attribute. When XLink resolution is triggered (usually at MPD load time), the client will query the URL referenced by this attribute. What happens next depends on the result of this query:
+
+<dl class="switch">
+
+: Non-empty result
+:: The entire XLink element is replaced with the query result. A single XLink element MAY be replaced with multiple elements of the same type.
+
+: Empty result or query failure
+:: The XLink element remains as-is.
+
+</dl>
+
+Recursive XLink elements SHALL NOT be used by services. Clients SH
```

Comment:
On a resolution additional `onLoad` attributes could result in indefinite resolution, but additional `onRequest` attributes can be used to allow for differentiated resolutions on repeat playouts. I believe the later is DASH specification compliant and is something we've actually used internally.

For our stack, we have one empty Period with an `onRequest`, as the player approaches the period it resolves the XLink and gets an ad period with additional `onRequest` attributes. The resolved period replaces the empty period and playout of the resolved period occurs. If the player is seeked back to before the ad period and approaches it again, it will then follow the additional `onRequest` to get a fresh ad experience.

The restriction in this scenario is that the XLink must resolve to only one period element as trying to resolve one to many and then later re-resolve and replace the many is too complicated to track player wise without introducing a mechanism to allow a resolution to consume adjacent elements.

Reply:
The motivation behind writing this sentence seems to be that I heard in one of the IOP calls the statement that "I think we forbid recursive xlink in IOP". If you say this is not right, I will just remove this sentence to avoid creating confusion. I was unable to find a specific statement in IOP so I will file a new issue for it.

Reply:
#267

Reply:
Marking this as resolved here with the removal of the sentence.